tap_google_ads/streams.py

```diff
@@ -172,11 +172,11 @@ def filter_out_non_attribute_fields(fields):
 
 class BaseStream:  # pylint: disable=too-many-instance-attributes
 
-    def __init__(self, fields, google_ads_resource_names, resource_schema, primary_keys):
+    def __init__(self, fields, google_ads_resource_names, resource_schema, primary_keys, automatic_keys = None):
         self.fields = fields
         self.google_ads_resource_names = google_ads_resource_names
         self.primary_keys = primary_keys
-
+        self.automatic_keys = automatic_keys if automatic_keys else set()
```

Comment:
We do this check later on
```python
if transformed_field_name in ({"date"} | self.automatic_keys):
```

which makes me think `"date"` is an automatic key. So I think we should make the following change 

```suggestion
        self.automatic_keys = {"date"}.union(automatic_keys) if automatic_keys else {"date"}
```

And notice the set literals (`{"date"}`) because `set("date")` would give you a set of characters I think

Reply:
This is used by core streams and report streams and we can't segment core streams by date. Gonna revert this.